addons/maintenance/models/maintenance.py

```diff
@@ -202,17 +204,53 @@ def _read_group_category_ids(self, domain, read_group_order=None, access_rights_
             fold[category.id] = category.fold
         return result, fold
 
+    def _create_new_request(self, date):
+        self.ensure_one()
+        self.env['maintenance.request'].create({
+                    'name': _('Preventive Maintenance - %s') % self.name,
+                    'request_date': date.strftime(DEFAULT_SERVER_DATE_FORMAT),
+                    'schedule_date': date.strftime(DEFAULT_SERVER_DATE_FORMAT),
+                    'category_id': self.category_id.id,
+                    'equipment_id': self.id,
+                    'maintenance_type': 'preventive',
+                    })
+
+    def _fill_up(self, from_date, to_date, to_end=False):
+        """
+            to_end takes into account that the distance to to_end is period also
+        """
+        self.ensure_one()
+        period = self.period
+        current_date = from_date + relativedelta(days=period)
+        while (current_date < to_date):
+            if to_end or current_date + relativedelta(days=period) < to_date:
+                self._create_new_request(current_date)
+            current_date = current_date + relativedelta(days=period)
```

Comment:
Couldn't that use `dateutil.rrule.rrule`? It works pretty well to iterate on dates.